lib/queueservice.js

```diff
@@ -0,0 +1,82 @@
+import xml2js from 'xml2js';
+import Debug from 'debug';
+import request from 'superagent-promise';
+
+let debug = Debug('taskcluster-docker-worker:queueService');
+
+export default class TaskQueue {
+  constructor(config) {
+    this.queues;
+    this.workerType = config.workerType;
+    this.provisionerId = config.provisionerId;
+    this.client = config.queue;
+    this.log = config.log;
+  }
+
+  async getQueues() {
+    if (!this.queues || (new Date(this.queues.expires)).getTime() > Date.now()) {
+      this.queues = await this.client.pollTaskUrls(this.provisionerId, this.workerType);
+    }
+    return this.queues.queues;
+  }
+
+  async getTasksFromQueue(queue, numberOfTasks) {
```

Comment:
The thought I was having here is getting the queues and querying the queues were two very dependent, often sequential, actions and didn't rely on some internal state of that class (list of queues).  Basically I want to know what queues exist, it tells me, and then for each of those I want to do something with them. If the task listener wants to log something about the queues, it already has the information because it has a queue object rather than some integer representing some state within this class that may or may not be the same depending on integer and when the queue refresh happens.
Definitely would need some handling around higher integer given but a lower number of queues because of a refresh.  If there is the possibility of the number of queues going down on a refresh, does the possibility also exist for the number to go up?  then the tasklistener wouldn't know about those until the next call to getTasksFromQueue..